File: z-clients/python/diom/models/msg_stream_seek_in.py
# this file is @generated

from ..internal.base_model import BaseModel
from ..internal.types import UnixTimestampMs

from .seek_position import SeekPosition


class MsgStreamSeekIn(BaseModel):
    namespace: str | None = None

    offset: int | None = None

    position: SeekPosition | None = None

    timestamp: UnixTimestampMs | None = None


class _MsgStreamSeekIn(BaseModel):
    namespace: str | None = None

    topic: str

    consumer_group: str

    offset: int | None = None

    position: SeekPosition | None = None

    timestamp: UnixTimestampMs | None = None

File: z-clients/javascript/src/models/msgStreamSeekIn.ts
// this file is @generated
import {
    type SeekPosition,
    SeekPositionSerializer,
} from './seekPosition';

export interface MsgStreamSeekIn {
    namespace?: string | null;
    offset?: number | null;
    position?: SeekPosition | null;
    timestamp?: Date | null;
}

export interface MsgStreamSeekIn_ {
    namespace?: string | null;
    topic: string;
    consumerGroup: string;
    offset?: number | null;
    position?: SeekPosition | null;
    timestamp?: Date | null;
}

export const MsgStreamSeekInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgStreamSeekIn_ {
        return {
            namespace: object['namespace'],
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            offset: object['offset'],
            position: object['position'] != null ? SeekPositionSerializer._fromJsonObject(object['position']): undefined,
            timestamp: object['timestamp'] ? new Date(Number(object['timestamp'])) : null,
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgStreamSeekIn_): any {
        return {
            'namespace': self.namespace,
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'offset': self.offset,
            'position': self.position != null ? SeekPositionSerializer._toJsonObject(self.position) : undefined,
            'timestamp': self.timestamp != null ? self.timestamp.getTime() : undefined,
        };
    }
}